dashboard/app/controllers/registrations_controller.rb

```diff
@@ -53,13 +59,16 @@ def begin_sign_up
     @user = User.new(begin_sign_up_params)
     @user.validate_for_finish_sign_up
     SignUpTracking.log_begin_sign_up(@user, session)
+    is_signup_post_enabled = DCDO.get('student-email-post-enabled', false)
 
     if @user.errors.blank?
       PartialRegistration.persist_attributes(session, @user)
-      redirect_to new_user_registration_path
+      redirect_to new_user_registration_path unless is_signup_post_enabled
     else
-      render 'new' # Re-render form to display validation errors
+      render 'new' unless is_signup_post_enabled # Re-render form to display validation errors
```

Comment:
In that case, why do we need the else? If the conditional is removed from line 71, it should handle the case where there are errors and `is_signup_post_enabled` is false.

Reply:
Oh I didn't catch that, good to know. That looks a lot cleaner with the `and return` instead of the else! It's probably a PEBKAC situation, but I find it hard to read Ruby control flow that includes if/elses with `unless` and `if` keywords inside.